config/routes.rb

```diff
@@ -3,4 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :dishes, only: :show do
+    resources :ingredients, only: :create, controller: :dish_ingredients
+  end
+
+  resources :chefs, only: :show do
+    resources :ingredients, only: :index, controller: "chefs/ingredients"
+  end
```

Comment:
These routes are great. It little bit of inconsistency with how you are declaring your "joins" controller, but that's just a minor callout.